File: prisma/seed.ts
import { PrismaClient } from '@prisma/client'
import { PrismaPg } from '@prisma/adapter-pg'
import { Pool } from 'pg'
import { config } from 'dotenv'
import bcrypt from 'bcryptjs'

config({ path: '../../.env' })

const connectionString = process.env.DATABASE_URL
const pool = new Pool({ connectionString })
const adapter = new PrismaPg(pool)

const prisma = new PrismaClient({ adapter })

async function main() {
  // Hash password for super admin
  const superAdminPassword = await bcrypt.hash('superadmin123', 10)

  // Create super admin user (no tenant)
  const superAdmin = await prisma.user.upsert({
    where: { email: 'superadmin@example.com' },
    update: {},
    create: {
      email: 'superadmin@example.com',
      name: 'Super Administrator',
      password: superAdminPassword,
      role: 'super_admin',
      tenantId: null, // Super admin has no tenant
    },
  })

  console.log('Seed data created:', { superAdmin })
}

main()
  .catch((e) => {
    console.error(e)
    process.exit(1)
  })
  .finally(async () => {
    await prisma.$disconnect()
  })

File: prisma/seed.py
import asyncio
import os
from dotenv import load_dotenv
from prisma import Prisma
import bcrypt

load_dotenv('../../.env')

async def main():
    db = Prisma()
    await db.connect()

    # 1. 准备加密密码
    password = "superadmin123"
    hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
    email = "superadmin@example.com"

    # 2. 模拟 upsert 逻辑 (先查后增)
    user = await db.user.find_unique(where={'email': email})

    if not user:
        new_user = await db.user.create(
            data={
                'email': email,
                'name': 'Super Administrator',
                'password': hashed_password,
                'role': 'super_admin',
                'tenantId': None,
            }
        )
        print(f"Seed data created: {new_user.name}")
    else:
        print("Super admin already exists, skipping...")

    await db.disconnect()

if __name__ == '__main__':
    asyncio.run(main())